xcell/cls/cl.py

```diff
@@ -226,12 +226,31 @@ def get_cl_file(self):
                 cl -= nl
             # Note that while we have subtracted the noise
             # bias from `cl_cp`, `cl_cov_cp` still includes it.
+            correction = None
+            if (mapper1.mask_power > 1) or (mapper2.mask_power > 1):
+                # Applies correction factor if masks have been
+                # implicitly applied to the maps
+                # See ACTk for reference
+                n_a = mapper1.mask_power
+                n_b = mapper2.mask_power
+                w_a = mapper1.get_mask()
+                w_b = mapper2.get_mask()
+                # Currently this first step is done with the pixell masks
+                correction = np.mean(w_a*w_b)/np.mean(w_a**n_a*w_b**n_b)
+                # Apply correction to all Cl's
+                cl *= correction
+                cl_cp *= correction
+                cl_cov_cp *= correction
+                cl_cov_11_cp *= correction
+                cl_cov_12_cp *= correction
+                cl_cov_22_cp *= correction
+
             np.savez(fname, ell=ell, cl=cl, cl_cp=cl_cp, nl=nl,
                      nl_cp=nl_cp, cl_cov_cp=cl_cov_cp,
                      cl_cov_11_cp=cl_cov_11_cp,
                      cl_cov_12_cp=cl_cov_12_cp,
                      cl_cov_22_cp=cl_cov_22_cp,
-                     wins=wins)
+                     wins=wins, correction=correction)
```

Comment:
Why do you want to store the correction?

Reply:
It was a debugging tool. 
I removed it.